code/functions/CMakeLists.txt

```diff
@@ -7,7 +7,10 @@ project( functions LANGUAGES CXX )
 include( "${CMAKE_CURRENT_SOURCE_DIR}/../CompilerSettings.cmake" )
 
 # Create the user's executable.
-add_executable( functions "Structs.h" "Structs.cpp" "functions.cpp" )
-if( "${CMAKE_CXX_COMPILER_ID}" STREQUAL "MSVC" )
-   target_compile_definitions( functions PRIVATE _CRT_SECURE_NO_WARNINGS )
+add_executable( functions Structs.cpp functions.cpp )
```

Comment:
When you remove headers from a target, they are no longer visible in IDE projects, like in Visual Studio (maybe also xcode and others).
Also, the `_CRT_SECURE_NO_WARNINGS` is necessary, because the example uses `strncpy`, which MSVC warns agains:
```
C:\dev\cpluspluscourse\code\functions\Structs.cpp(10,5): warning C4996: 'strncpy': This function or variable may be uns
afe. Consider using strncpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details
. [C:\dev\cpluspluscourse\code\functions\build\functions.vcxproj]
C:\dev\cpluspluscourse\code\functions\Structs.cpp(15,5): warning C4996: 'strncpy': This function or variable may be uns
afe. Consider using strncpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details
. [C:\dev\cpluspluscourse\code\functions\build\functions.vcxproj]
```

Reply:
You reviewed faster than me ;)
A bad rebasing, I guess.
I remove this change.